data/src/test/java/org/apache/iceberg/TestScanPlanningAndReporting.java

```diff
@@ -0,0 +1,334 @@
+    Table tbl =
+        TestTables.create(
+            tableDir,
+            "scan-planning-with-deletes",
+            SCHEMA,
+            SPEC,
+            SortOrder.unsorted(),
+            formatVersion,
+            reporter);
+
+    tbl.newAppend().appendFile(FILE_A).appendFile(FILE_B).appendFile(FILE_C).commit();
+    tbl.newRowDelta().addDeletes(FILE_A_DELETES).addDeletes(FILE_B_DELETES).commit();
+    TableScan tableScan = tbl.newScan();
+
+    try (CloseableIterable<FileScanTask> fileScanTasks = tableScan.planFiles()) {
+      fileScanTasks.forEach(task -> {});
+    }
+
+    ScanReport scanReport = reporter.lastReport();
+    assertThat(scanReport).isNotNull();
+    assertThat(scanReport.tableName()).isEqualTo("scan-planning-with-deletes");
+    assertThat(scanReport.snapshotId()).isEqualTo(2L);
+    assertThat(scanReport.scanMetrics().totalPlanningDuration().totalDuration())
+        .isGreaterThan(Duration.ZERO);
+    assertThat(scanReport.scanMetrics().resultDataFiles().value()).isEqualTo(3);
+    assertThat(scanReport.scanMetrics().resultDeleteFiles().value()).isEqualTo(2);
+    assertThat(scanReport.scanMetrics().scannedDataManifests().value()).isEqualTo(1);
+    assertThat(scanReport.scanMetrics().skippedDataManifests().value()).isEqualTo(0);
+    assertThat(scanReport.scanMetrics().totalDataManifests().value()).isEqualTo(1);
+    assertThat(scanReport.scanMetrics().totalDeleteManifests().value()).isEqualTo(1);
+    assertThat(scanReport.scanMetrics().totalFileSizeInBytes().value()).isEqualTo(30L);
+    assertThat(scanReport.scanMetrics().totalDeleteFileSizeInBytes().value()).isEqualTo(20L);
+  }
+
+  @Test
+  public void multipleDataManifests() throws IOException {
+    Table tbl =
+        TestTables.create(
+            tableDir,
+            "multiple-data-manifests",
+            SCHEMA,
+            SPEC,
+            SortOrder.unsorted(),
+            formatVersion,
+            reporter);
+
+    tbl.newAppend().appendFile(FILE_A).appendFile(FILE_B).commit();
+    long snapshotIdOne = tbl.currentSnapshot().snapshotId();
+    tbl.newAppend().appendFile(FILE_C).appendFile(FILE_D).commit();
+    long snapshotIdTwo = tbl.currentSnapshot().snapshotId();
+
+    List<ManifestFile> manifests = tbl.currentSnapshot().dataManifests(tbl.io());
+
+    ManifestFile manifestOne =
+        writeManifest(
+            "manifest-file-1.avro",
+            manifestEntry(Status.EXISTING, snapshotIdOne, FILE_A),
+            manifestEntry(Status.EXISTING, snapshotIdOne, FILE_B));
+
+    ManifestFile manifestTwo =
+        writeManifest(
+            "manifest-file-2.avro",
+            manifestEntry(Status.EXISTING, snapshotIdTwo, FILE_C),
+            manifestEntry(Status.EXISTING, snapshotIdTwo, FILE_D));
+
+    RewriteManifests rewriteManifests = tbl.rewriteManifests();
+    manifests.forEach(rewriteManifests::deleteManifest);
+    rewriteManifests.addManifest(manifestOne).addManifest(manifestTwo);
+    rewriteManifests.commit();
```

Comment:
There's no need to use `rewriteManifests`. Each fast append will create a new manifest file, so I'd just use that instead.

We also need a test case that filters manifests and results in `skippedDataManifests` with a non-zero value.

Reply:
updated. Also the second part in that test method has `assertThat(scanReport.scanMetrics().skippedDataManifests().value()).isEqualTo(1)`